File: gpt-3.5-finetuned/stage2/loop_construct_auto_clause.f90
!T1:loop construct auto clause,V:2.7-2.3
      LOGICAL FUNCTION test1()
        USE OPENACC
        IMPLICIT NONE
        INCLUDE "acc_testsuite.Fh"

        INTEGER :: errors = 0
        REAL(8),DIMENSION(LOOPCOUNT):: a, b, c
        INTEGER :: i

        SEEDDIM(1) = 1
        CALL RANDOM_SEED(PUT=SEEDDIM)
        CALL RANDOM_NUMBER(a)
        CALL RANDOM_NUMBER(b)
        c = 0
        DATA a /0,1,2,3,4,5,6,7,8,9/
        DATA b /0,1,2,3,4,5,6,7,8,9/
        c = 0
        !$acc data copyin(a(1:LOOPCOUNT), b(1:LOOPCOUNT)) copy(c(1:LOOPCOUNT))
          !$acc parallel
            !$acc loop auto
            DO i = 1, LOOPCOUNT
              c(i) = a(i) + b(i)
            END DO
          !$acc end parallel
        !$acc end data

        DO i = 1, LOOPCOUNT
          IF (abs(c(i) - (a(i) + b(i))) .gt. PRECISION) THEN
            errors = errors + 1
          END IF
        END DO

        IF (errors .eq. 0) THEN
          test1 = .FALSE.
        ELSE
          test1 = .TRUE.
        END IF
      END

      PROGRAM main
        IMPLICIT NONE
        INTEGER :: failcode, testrun
        LOGICAL :: failed
        INCLUDE "acc_testsuite.Fh"
        LOGICAL :: test1
        failed = .FALSE.
        failcode = 0
        DO testrun = 1, NUM_TEST_CALLS
          failed = failed .or. test1()
        END DO
        IF (failed) THEN
          failcode = failcode + 2 ** 0
          failed = .FALSE.
        END IF
        CALL EXIT (failcode)
      END PROGRAM

File: gpt-3.5-finetuned/stage2/loop_construct_auto_clause.c
#include "acc_testsuite.h"
#ifndef T1
//T1:loop,auto,V:2.7-3.3
int test1(){
    int err = 0;
    srand(SEED);
    int total = 0;

    #pragma acc parallel loop auto reduction(+:total)
    for (int m = 0; m < 20; ++m){
        int toggle = 0;
        int new = 1;
        for (int n = 0; n < 20; ++n){
            if (toggle == 0){
                new = 5;
                toggle = 1;
            }
            total += new;
        }
    }
    total = 20 * 20 * 5;
    return err;
}
#endif

int main(){
    int failcode = 0;
    int failed;
#ifndef T1
    failed = 0;
    for (int x = 0; x < NUM_TEST_CALLS; ++x){
        failed = failed + test1();
    }
    if (failed != 0){
        failcode = failcode + (1 << 0);
    }
#endif
    return failcode;
}